src/api-client.ts

```diff
@@ -65,6 +65,12 @@ export class ApiClient {
         if (idempotencyKey) {
             headers["Idempotency-Key"] = idempotencyKey;
         }
+
+        const ncwWalletId = requestOptions?.NCW?.walletId;
+        if (ncwWalletId) {
+            headers["X-End-User-Wallet-Id"] = ncwWalletId;
+        }
```

Comment:
ApiClient class is a generic HTTP client wrapper and shouldn't include business logic related functionality

Reply:
where do you see business logic here? it's request options mapped to an http header

Reply:
By business logic I mean that the ApiClient shouldn't be aware of the NCW term.
The `idempotencyKey` is a generic header that affects all endpoints, but I still agree.
Ideally the `headers` should have been given as an optional argument, which could be set in method caller, with the final `Authorization` header addition.

Reply:
`X-End-User-Wallet-Id` is a generic header is user together with `Authorization` header generically (recent additions)